src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,201 @@
+        int taskNumber = Integer.parseInt(taskItemNumber);
+        if (taskList.get(taskNumber - 1) == null) {
+            throw new DukeException(
+                    "____________________________________________________________\n"
+                            + "☹ OOPS!!! The task you chose to delete does not exist."
+                            + " Use the 'list' command to check the items in your list.\n"
+                            + "____________________________________________________________"
+            );
+        }
+        taskList.get(taskNumber - 1).completeTask();
+        return Ui.completedTaskMessage(taskList.get(taskNumber - 1));
+    }
+
+    /**
+     * addToList adds a taskItem of type duke.TaskItem to the list.
+     * @param taskItem a duke.TaskItem that is to be added to duke.Duke's list.
+     */
+    public static String addToList(TaskItem taskItem) {
+        Duke.taskList.add(taskItem);
+        Duke.listIndex = taskList.size();
+        String body = "";
+        /*
+        System.out.println("____________________________________________________________");
+        System.out.println("Got it. I've added this task:");
+        System.out.println(taskItem.toString());
+        */
+        if (listIndex == 1) {
+            /* System.out.println("Now you have " + 1 + " task in the list."); */
+            body = "Now you have " + 1 + " task in the list.\n";
+        }
+        if (listIndex > 1) {
+            /* System.out.println("Now you have " + (Duke.listIndex) + " tasks in your list."); */
+            body = "Now you have " + (Duke.listIndex) + " tasks in your list.\n";
+        }
+        return "____________________________________________________________\n"
+                + "Got it. I've added this task:\n"
+                + taskItem.toString() + "\n"
+                + body
+                + "____________________________________________________________";
+    }
+
+    /**
+     * prints the list upon inputting the 'list' command.
+     */
+    public static String printList() {
+        int number = 1;
+        String output = "____________________________________________________________\n"
+                + "Here are the tasks in your list:\n";
+        /*
+        System.out.println("____________________________________________________________");
+        System.out.println("Here are the tasks in your list:");
+        */
+        for (int i = 0; i < taskList.size(); i++) {
+            if (taskList.get(i) != null) {
+                output += number + "." + taskList.get(i).toString() + "\n";
+                number++;
+            }
+        }
+        return output;
+    }
+
+    /**
+     * Finds tasks based on the keyword the user inputs and prints them to the user
+     * @param keyword keyword used to find matching tasks
+     * @return a TaskList with matching tasks to be printed.
+     */
+    public static TaskList findTask(String keyword) {
```

Comment:
As mentioned earlier, you may want to consider adding all the methods related to the task list (adding, deleting, marking and finding tasks as well as printing the list) under Tasklist.java instead?